src/seacoral-cbmc/results.mli

```diff
@@ -23,19 +25,25 @@ val get_covered : t -> Basics.Ints.t
     returns the empty set. *)
 val get_uncoverable : t -> Basics.Ints.t
 
-(** Reads the result of a CBMC cover analysis. *)
-val goals_to_test_cases :
-  env:Types.simple_label_env ->
-  harness:Harness.t ->
-  Types.DATA.cbmc_cover_output Types.DATA.cell list ->
-  t
+(** [goal_stream_to_test_cases ~env ~harness ~stream kont]
+
+    Reads the result of a CBMC cover analysis and calls [kont] on the
+    result. *)
+val goal_stream_to_test_cases :
+  env:Types.simple_label_env
+  -> harness:Harness.t
+  -> stream:Types.DATA.cbmc_cover_output Types.DATA.cell Lwt_stream.t
+  -> ((Sc_values.literal_binding * Basics.Ints.t) list -> unit Lwt.t)
+  -> t Lwt.t
```

Comment:
I don't really understand why there is a need to pass a `kont` that necessarily hides side effects. Maybe this should use a `Lwt_stream.map_filter` internally to return a `(Sc_values.literal_binding * Basics.Ints.t) list Lwt_stream.t`, or something similar, directly.

Reply:
Just that the signature forces the use of a side-effect. Returning a stream would be cleaner and offer more options for processing the results.